synapse/cores/common.py

```diff
@@ -1799,6 +1800,10 @@ def addTufoEvents(self, form, propss):
                     xact.fire('node:add', form=form, valu=valu, node=node)
                     xact.spliced('node:add', form=form, valu=valu, props=props)
 
+                if self.autoadd:
```

Comment:
@vEpiphyte - this "ro props" change looks good to me. If you want, we can merge this now as a separate PR.